tests/test_tasks.py

```diff
@@ -860,13 +873,20 @@ def my_flow():
         with pytest.raises(ValueError, match="deadlock"):
             my_flow()
 
-    def test_logs_message_when_submitted_tasks_end_in_pending(self, caplog):
+    def test_logs_message_when_submitted_tasks_end_in_pending(
+        self, caplog, enable_client_side_task_run_orchestration
+    ):
         """
         If submitted tasks aren't waited on before a flow exits, they may fail to run
         because they're transition from PENDING to RUNNING is denied. This test ensures
         that a message is logged when this happens.
         """
 
+        if enable_client_side_task_run_orchestration:
+            pytest.xfail(
```

Comment:
This test cannot pass without the behavior of `PreventRunningTasksFromStoppedFlows`. I am not sure how to interpret this test.

With server side orchestration:
- the flow run goes to `RUNNING`
- the task run is submitted
- the flow run goes to `COMPLETED`
- the task run attempts to go to `RUNNING` and is rejected because the flow is completed and stays in `PENDING`

With client side orchestration:
- the flow run goes to `RUNNING`
- the task run is submitted
- the task run goes to `RUNNING` and begins to execute since we have no parallel to `PreventRunningTasksFromStoppedFlows`

I don't think this is a critical test, so I've marked it as `xfail` for the time being. However, the concern I have here is that the flow run going to `COMPLETED` _first_ seems to be a byproduct of the latency of the task run needing to go the API?

Reply:
Yup this is fine. This can only really happen when tasks are running in distributed contexts and for that, I think the better solution (if it becomes an issue) will be tighter integrations, not relying on orchestration rules.